Allow using JDKs without bundled JavaFX

Katharsas opened the pull request:
**I need somebody with intelliJ Ultimate to test the run configurations.**

Changes the main class (entrypoint) so that it no longer is the same class that extends Application.
- New main method inside com.faforever.client.Main, updated all references
- Explicitly sets entry point for bootRun in build.gradle
- Removes additional ambigous main method from WindowsUtil (strictly this only needed if entry point for boot run is not set, but i guess i remove this just in case)
- Renames original main method to avoid confusion

Workaround for #1298

1-alex98 commented:
So u ran a project wide search for the Main class and replaced it?

Katharsas commented:
Yeah basically. Also since i renamed the original main method, if this breaks anything, it should break pretty obviously. At least i hope so^^

Brutus5000 commented:
Does this enable Java 11 usage?

Katharsas commented:
Yes. Last time i tested it worked fine with AdoptOpenJDK 10 and 11

Geosearchef reviewed (dismissed):
I don't like this, but I've stumbled over that problem myself and haven't found any other solution aswell.

Brutus5000 reviewed (commented):
This throws annoying IntelliJ Ultimate errors: `Run Configuration Error: Not a valid Spring Boot application class` (I can still run it though)

Katharsas commented:
Are you ok with that? Since i don't have Intellij Ultimate i cannot test what changes to run configurations do.

Can we maybe just remove the run configurations alltogether and instead put the VM parameters into the gradle file?^^ Imo profiles should generally be part of the IDE-independent gradle build anyway. I don't see a reason why we should use Intellij specific profile definitions.

1-alex98 commented:
Well the think is the Main class is no longer a Spring Application thingy we should change that indeed

heelix commented:
I've got ultimate, if you are lookin g for a test monkey.

On Sat, Jul 13, 2019 at 8:40 AM Alexander von Trostorff <
notifications@github.com> wrote:

> Well the think is the Main class is no longer a Spring Application thingy
> we should change that indeed
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/FAForever/downlords-faf-client/pull/1325?email_source=notifications&email_token=AAQDGTA5SHUNNBJW5LYZOO3P7HLMRA5CNFSM4H4JKXE2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODZ3R4JY#issuecomment-511122983>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAQDGTEEJUJCKZ5ZCLDPO2LP7HLMRANCNFSM4H4JKXEQ>
> .
>

Katharsas commented:
Well i can just copy the `@SpringBootApplication` annotation to the new main class and HOPE that that fixes it and doesn't break anything else.

Katharsas committed:
Speculate IntelliJ error fix

1-alex98 commented:
Yeah Brutus might also know what is needed for a Spring Application

Katharsas commented:
@Brutus5000 can you check if you still get the error?

Katharsas commented:
@heelix You could try if you get the problem described by Brutus or if it was fixed.

Brutus5000 commented:
@Katharsas IntelliJ errors have vanished

Katharsas committed:
Allow using JDKs without bundled JavaFX

Changes the main class (entrypoint) so that it no longer is the same class that extends Application.
- New main method inside com.faforever.client.Main, updated all references
- Explicitly sets entry point for bootRun in build.gradle
- Removes additional ambigous main method from WindowsUtil (strictly this only needed if entry point for boot run is not set)
- Renames original main method to avoid confusion
Workaround for #1298

Katharsas commented:
Same error happens on Linux as well (at least for me on Ubuntu LTS after installing JDK 11), and we don't ship a JDK with the Linux install, so im in favour of getting this merged finally.
@axel1200

Brutus5000 commented:
I'd be in favor of merging it *after* releasing 1.0

Katharsas commented:
> 
> 
> I'd be in favor of merging it _after_ releasing 1.0

I don't agree.
I was trying to fix #1462 and already would have needed this PR twice just for that issue (1. because sometimes i need to use gradle without an IDE for example for debugging and 2. because i need to test later JavaFX versions to check if they fix/change the Tooltip implementation).
Sorry, but it is becoming too frustrating to always apply this PR onto anything i do so i have stopped any work on this Repo now until this is merged.
